### DISCUSSION TITLE
Duplicate item button

### DISCUSSION THREAD

**Discussion by abanchev** [strapi/strapi#7646] [Legacy Topics]
Is there a plan to have a duplicate item button when one is selected in the collection list.
It should just open the editor for a new item, with the data of the selected item pre-filled.

**Comment by derrickmehaffy** [comment]
@Aurelsicoko @JabStrapi

**Reply by lauriejim** [reply]
Adding a note that I already see this request during pre-sale calls.
Could be really cool to have this button close to the edit and delete icon (same as we did for permissions)

**Reply by derrickmehaffy** [reply]
I agree @lauriejim and it shouldn't be too difficult to implement.

**Reply by tsvecak** [reply]
Is this something that has to go onto roadmap or a feature that might be silently added? 🙂

**Reply by derrickmehaffy** [reply]
Most likely silently added :P

**Reply by tsvecak** [reply]
Any ETA on that or still in the air? 🙂

**Reply by derrickmehaffy** [reply]
Pretty much still in the air at this moment unless someone in the community feels like making a pull request :wink:

**Reply by MeXaaR** [reply]
I made a PR on this, maybe there is some changes to do but it seems to work #8085

**Comment by sam-pires** [comment]
While reviewing @MeXaaR PR #8563 (thanks again!!), we wrote the full specifications for this feature. I share it here so we keep a history of it:

### Specs

## Description
This button is to be added on the right side of each row, in the CM list views. By clicking on it, a user can duplicate the targeted content.

## Design specifications
_⚠️ FYI -  No specs were shared before the PR got submitted, these ones have been added after the review of #8563_ 

- The icon and its position should follow the same as in the RBAC role management view (see screenshot below)
  - Duplicate button must be located on the left of the pen.
  - The duplicate icon must be the same as the one in the role list.
<img width="902" alt="Capture d’écran 2020-11-04 à 18 27 06" src="https://user-images.githubusercontent.com/25405887/98689053-cea00b00-236b-11eb-9657-1c2bcba12c2b.png">


## Management rule

- When duplicating an entity:
    - The fields of the duplicated entity must all be filed with the original content. Only the ID differs and all `created`keys are deleted.
    - In case of a "unique" validation on a field, the content of the duplicate is filled with the original content. (The error validation will appear when the user tries to save, that's fine)

- If Draft&Published is enabled, a content duplicated from a `Published` entity, should be created as `Draft`

- The duplicate button must appear on all List views of the CM, but RBAC permissions must apply to it. So if a user doesn't have the right to create a content, he/she won't see the button.

## Acceptance tests

- [ ]  As an Admin panel user
    - [ ]  when I have rights to create content
        - [ ]  then I can see the duplicate icon on the list view
        - [ ]  then I can duplicate the content from the list view
            - [ ]  When I duplicate a content then
                - [ ]  I automatically go to the edit view of the duplicated content
                - [ ]  the field of the duplicated content are filled with the original content
                - [ ]  the duplicated content's ID is different than the original one
                - [ ]  the `create`  keys are empty
                - [ ]  If the Draft&Published feature is enabled for this CT, then when I duplicate a content:
                    - [ ]  No matter what's the original content's status, the duplicated content must be a draft
    - [ ]  when I don't have rights to create for the content
        - [ ]  then I do not see the duplicate icon in the list view


FYI - The acceptance tests are here to guide the developer but they will be checked by the person doing the QA on the feature (from the Strapi team)
